### DISCUSSION TITLE
How do open local window choose files or folder in slint?

### DISCUSSION THREAD

**Discussion by joyoes** [slint-ui/slint#3015] [General] [answered]
How do open local window choose files or folder in slint?
Please provide an example of implementation
Thanks

**Comment by bjorn** [comment]
I don't believe Slint provides a native file dialog API at the moment, so you'll need to use another library for this, depending on whether you're using Rust, C++, etc. Assuming you're using Rust, you could for example use https://crates.io/crates/native-dialog (I haven't tested it and didn't do any kind of comparison between available options).

**Reply by TillerBelotti** [reply, marked as answer]
The native-dialog crate works perfectly alongside Slint, I use it to display alerts and the file selector in various projects.
